File: src/playwright/waits.ts
import type { Page } from 'playwright';

export async function waitForInitialLoad(page: Page): Promise<void> {
  await page.waitForLoadState('domcontentloaded');
  await Promise.race([
    page.waitForLoadState('networkidle', { timeout: 3000 }),
    new Promise<void>((resolve) => setTimeout(resolve, 500))
  ]);
}

File: browser_platform/playwright/waits.py
from __future__ import annotations
import asyncio
from playwright.async_api import Page


async def wait_for_initial_load(page: Page) -> None:
    await page.wait_for_load_state('domcontentloaded')
    try:
        await asyncio.wait_for(
            asyncio.shield(asyncio.ensure_future(page.wait_for_load_state('networkidle'))),
            timeout=3.0
        )
    except (asyncio.TimeoutError, Exception):
        pass
